Enable searching of extracted text

awead opened the pull request:
Returns a work if any of its associated files contain text that matches the user's query. It uses a Solr join to link the work to the file resource in Solr, which uses the Lucene query parser and not the default edismax one.

Fixes #1025

mtribone commented:
![screen shot 2017-10-24 at 3 44 46 pm](https://user-images.githubusercontent.com/4163828/31964618-a0b698fa-b8d2-11e7-920b-0ef25c3a87ed.png)

cam156 commented on spec/features/generic_work/work_with_many_files_spec.rb:
```diff
@@ -47,10 +47,13 @@
           expect(page).to have_link("File #{id}")
         end
       end
+      within('div.pager') do
```
@mtribone We may want to click on the page link and check the next page to see if files 11..20 exists on the second page.
Additionally will there be other page links? 2..10?

mtribone replied:
Yes there will be other page links. Should we test each one or is the presence of a second page _with_ records enough?

cam156 replied:
@mtribone testing that the links exist and that we can get to the second page should be enough.  @awead do you agree?

awead replied:
@mtribone yeah. Also, checking for the link with `have_link` and then clicking it are the same thing. You could probably just say `click_link`. If it's not there or it can't find, it'll break.

awead commented on app/views/curation_concerns/base/_paginate.html.erb:
```diff
@@ -0,0 +1,6 @@
+<% if @presenter.member_presenters.total_pages > 1 %>
```
@mtribone is this a partial you're overriding? Can you put in `<%# comment ... %>` on the first line about why?

mtribone replied:
Okay. I believe that I had to pinch the paginate from Curation Concerns collections and use the presenter because the fieldsets are ordered. Whereas the collection listing of works is a Solr response.
https://github.com/samvera/curation_concerns/blob/master/app/views/collections/_paginate.html.erb

mtribone replied:
And give it a new method of `file_page` for page in the work_show_presenter

mtribone replied:
@awead @cam156 Does this comment make sense to you?

"Need to pinch the paginator from https://github.com/samvera/curation_concerns/blob/master/app/views/collections/_paginate.html.erb
    because the fieldsets for works are ordered and not a solr response. Uses the presenter and a new method of file_page to create pages of ordered objects"

awead replied:
@mtribone great!

awead commented on spec/features/generic_work/work_with_many_files_spec.rb:
```diff
@@ -42,15 +42,28 @@
+        click_link('2')
+      end
+      within('table.related-files') do
+        (2..11).each do |id|
```
Shouldn't this be `(11..20)`? I thought we were on the second page of 10 items out of 100.

mtribone replied:
I went to art school, so math isn't one of my subjects.

awead replied:
@mtribone I went to music school, so I can only count to 4. Don't trust my math either... `(2..11)` may in fact be correct. I would think that the test would fail, though, because you'd be looking at files 11 through 20. Try throwing in a `save_and_open_page` and just verify that you're seeing what you're supposed to be seeing.

mtribone replied:
@awead I guess the files were still within the range even though file 11 wasn't present. Save and open page does show me files 11-20.

awead replied:
@mtribone ok. I guess just update the test to check that range and we should be good.

awead committed:
Enable searching of extracted text

Returns a work if any of its associated files contain text that matches
the user's query. It uses a Solr join to link the work to the file
resource in Solr, which uses the Lucene query parser and not the default
edismax one.
